modules/SofaBoundaryCondition/SofaBoundaryCondition_test/FixedConstraint_test.cpp

```diff
@@ -156,6 +158,91 @@ struct FixedConstraint_test : public BaseTest
         return true;
 
     }
+
+    bool testTopologicalChanges()
+    {
+        simulation::Simulation* simulation;
+        sofa::simulation::setSimulation(simulation = new sofa::simulation::graph::DAGSimulation());
+        
+        /// Scene creation
+        simulation::Node::SPtr root = simulation->createNewGraph("root");
+        root->setGravity(defaulttype::Vector3(0, 0, 0));
+
+        /// Create euler solver
+        simulation::Node::SPtr node = createEulerSolverNode(root, "test");
+
+        /// Add mechanicalObject
+        sofa::Size nbrDofs = 5;
+        typename MechanicalObject::SPtr dofs = addNew<MechanicalObject>(node);
+        dofs->resize(nbrDofs);
+        
+        /// Add PointSetTopology
+        typename sofa::component::topology::PointSetTopologyContainer::SPtr tCon = sofa::core::objectmodel::New<sofa::component::topology::PointSetTopologyContainer>();
+        typename sofa::component::topology::PointSetTopologyModifier::SPtr tMod = sofa::core::objectmodel::New<sofa::component::topology::PointSetTopologyModifier>();
+        tCon->setNbPoints(nbrDofs);
+        node->addObject(tCon);
+        node->addObject(tMod);
+
+        /// Add Mass
+        createUniformMass<DataTypes>(node, *dofs.get());
+
+        /// Add force
+        typename ForceField::SPtr forceField = addNew<ForceField>(node);
+        
+        // create a force vector
+        Deriv force;
+        for (unsigned i = 0; i < force.size(); i++)
+            force[i] = 10;
+
+        /// Fill position and force
+        typename MechanicalObject::WriteVecCoord writeX = dofs->writePositions();
+        for (sofa::Index id = 0; id < nbrDofs; id++)
+        {
+            for (unsigned int i = 0; i < writeX[0].size(); i++)
```

Comment:
I assume it leads to the same result if you take `writeX[0].size()` or `writeX[id].size()`?